### PR TITLE
Upload crm report

### PR DESCRIPTION
**Pull request by BillAnastasiadis**
This pr enables the upload of the results of `crm report` from the cluster machines to openqa
- Related ticket: https://jira.suse.com/browse/TEAM-8138
- Verification run: http://openqaworker15.qa.suse.cz/tests/207356#downloads
(failed on purpose at the end of `deploy`)

### CODE REVIEW

File: lib/qesapdeployment.pm [hunk 1]
```diff
@@ -585,6 +595,76 @@ sub qesap_ansible_script_output {
...
+
+    qesap_ansible_fetch_file(provider => 'aws', host => 'vmhana01', root => 1);
+
+    It uses playbook data/sles4sap/fetch_file.yaml
+
+    1. ansible-playbook run the playbook
+    3. the playbook download the file locally
+    4. the file is read and stored to be returned to the caller
+
+=over 8
```
Comment (mpagot) [inline lib/qesapdeployment.pm:611]: 7 arguments

Comment (BillAnastasiadis) [reply lib/qesapdeployment.pm:613]: I had forgotten one, now 8

File: data/sles4sap/fetch_file.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+- hosts: all
+  vars:
+    path: '/tmp/'
```
Comment (mpagot) [inline data/sles4sap/fetch_file.yaml:4]: What about renaming `path` and `out_path` to `remote_path` and `local_path` ?

File: data/sles4sap/fetch_file.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    out_path: '/tmp/ansible_fetch_output/'
+    file: 'testout.txt'
+  tasks:
+  - name: fetch
+    fetch:
+      flat: yes
+      src: "{{ path }}{{ file }}"
+      dest: "{{ out_path }}"
```
Comment (mpagot) [inline data/sles4sap/fetch_file.yaml:12]: Please check if an empty line is needed at the end of this file

File: data/sles4sap/script_output.yaml [hunk 4]
```diff
@@ -2,16 +2,11 @@
 - hosts: all
   vars:
     cmd: ''
-    out_path: '/tmp/ansible_script_output/'
+    path: '/tmp/'
```
Comment (mpagot) [inline data/sles4sap/script_output.yaml:5]: What about renaming `path`  to `remote_path`  ?

File: lib/qesapdeployment.pm [hunk 5]
```diff
@@ -666,6 +750,39 @@ sub qesap_wait_for_ssh {
...
+=back
+=cut
+
+sub qesap_upload_crm_report {
+    my (%args) = @_;
+    croak 'Missing mandatory host argument' unless $args{host};
+    my $host = $args{host};
+    my $prov = get_required_var('PUBLIC_CLOUD_PROVIDER');
+    my $crm_log = "/var/log/$host-crm_report";
+    my $report_opt = !is_sle('12-sp4+') ? '-f0' : '';
```
Comment (mpagot) [inline lib/qesapdeployment.pm:770]: is there any way not to have this `is_sle` dependency?

File: lib/qesapdeployment.pm [hunk 6]
```diff
@@ -510,13 +511,13 @@ sub qesap_ansible_cmd {
 
     1. ansible-playbook run the playbook
```
Comment (mpagot) [inline lib/qesapdeployment.pm:512]: run`s`

**Review by lpalovsky — APPROVED** [review]
LGTM

File: t/09_qesapdeployment.t [hunk 7]
```diff
@@ -420,16 +420,17 @@ subtest '[qesap_ansible_script_output]' => sub {
...
             return 'ANEMONE' if ($_[0] =~ /cat.*/); });
+    $qesap->redefine(qesap_ansible_script_output_file => sub { return '/tmp/ansible_script_output/'; });
 
-    my $out = qesap_ansible_script_output(cmd => 'SWIM', provider => 'NEMO', host => 'REEF');
+    my $out = qesap_ansible_script_output(cmd => 'SWIM', provider => 'NEMO', host => 'REEF', file => 'testout.txt', out_path => '/tmp/ansible_script_output/');
 
     note("\n  out=$out");
     note("\n  C-->  " . join("\n  C-->  ", @calls));
-    ok((any { /ansible-playbook.*REEF.*"cmd='SWIM'"/ } @calls), 'proper ansible-playbooks command');
+    #ok((any { /ansible-playbook.*REEF.*"cmd='SWIM'"/ } @calls), 'proper ansible-playbooks command');
```
Comment (mpagot) [inline t/09_qesapdeployment.t:429]: why this commented line?

**Review by mpagot — APPROVED** [review]
LGTM

**Review by alvarocarvajald — COMMENTED** [review]
LGTM

**Pull request merged**